hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/ReadHandler.java

```diff
@@ -16,11 +16,17 @@
 
 package com.hazelcast.internal.tpc;
 
-/**
- * The ReadHandler currently is a mess. The problem is that there is currently not a unified buffer
- * that fits with Nio (ByteBuffer) and IOUring (Netty ByteBuf). So we need to fix the
- * {@link com.hazelcast.internal.tpc.iobuffer.IOBuffer} so that we can have a single
- * ReadHandler that can be used for NioEventloop/IOUringEventLoop/EpollEventloop.
- */
```

Comment:
What happened to this dilemma? Does Nio ByteBuffer meets all of our needs now?

Reply:
For the reading part the ByteBuffer is ok. 

The problem was that the io_uring implementation was relying on the Netty bytebuf and since I got rid of Netty, the io_uring reading side works based on the Nio ByteBuffer.

So for reading data from the socket the NIO ByteBuffer is ok; it it is just a pointer to some data. Once you start to process it, you need to move it into some specific like a ClientMessage,Packet, IOBuffer etc.